packages/@aws-cdk/aws-apigatewayv2/lib/http/vpc-link.ts

```diff
@@ -0,0 +1,127 @@
+  /**
+   * Physical ID of the VpcLink resource
+   * @attribute
+   */
+  readonly vpcLinkId: string;
+}
+
+/**
+ * Properties for a VpcLink
+ */
+export interface VpcLinkProps {
+  /**
+   * The name used to label and identify the VPC link.
+   * @default - automatically generated name
+   */
+  readonly vpcLinkName?: string;
+
+  /**
+   * The network load balancers of the VPC targeted by the VPC link.
+   * The network load balancers must be owned by the same AWS account of the API owner.
+   *
+   * @default - no targets. Use `addTargets` to add targets
+   */
+  readonly targets?: elbv2.INetworkLoadBalancer[];
+
+  /**
+   * A list of security groups for the VPC link.
+   *
+   * @default - no security groups. Use `addSecurityGroups` to add security groups
+   */
+  readonly securityGroups?: ec2.ISecurityGroup[];
+}
+
+
+/**
+ * Define a new VPC Link
+ * Specifies an API Gateway VPC link for a HTTP API to access resources in an Amazon Virtual Private Cloud (VPC).
+ */
+export class VpcLink extends Resource implements IVpcLink {
+  /**
+   * Import a VPC Link by its Id
+   */
+  public static fromVpcLinkId(scope: Construct, id: string, vpcLinkId: string): IVpcLink {
+    class Import extends Resource implements IVpcLink {
+      public vpcLinkId = vpcLinkId;
+    }
+
+    return new Import(scope, id);
+  }
+
+  /**
+   * Physical ID of the VpcLink resource
+   * @attribute
+   */
+  public readonly vpcLinkId: string;
+
+  private readonly targets = new Array<elbv2.INetworkLoadBalancer>();
+  private readonly securityGroups = new Array<ec2.ISecurityGroup>();
+
+  constructor(scope: Construct, id: string, props: VpcLinkProps = {}) {
+    super(scope, id, {
+      physicalName: props.vpcLinkName ||
+        Lazy.stringValue({ produce: () => this.node.uniqueId }),
+    });
+
+    const cfnResource = new CfnVpcLink(this, 'Resource', {
+      name: this.physicalName,
+      subnetIds: Lazy.listValue({ produce: () => this.renderTargets() }),
+      securityGroupIds: Lazy.listValue({ produce: () => this.renderSecurityGroups() }),
+    });
+
+    this.vpcLinkId = cfnResource.ref;
+
+    if (props.targets) {
+      this.addTargets(...props.targets);
+    }
+
+    if (props.securityGroups) {
+      this.addSecurityGroups(...props.securityGroups);
+    }
+  }
+
+  /**
+   * Adds the provided targets to the vpc link
+   *
+   * @param targets
+   */
+  public addTargets(...targets: elbv2.INetworkLoadBalancer[]) {
+    this.targets.push(...targets);
+  }
+
+  /**
+   * Adds the provided security groups to the vpc link
+   *
+   * @param groups
+   */
+  public addSecurityGroups(...groups: ec2.ISecurityGroup[]) {
+    this.securityGroups.push(...groups);
+  }
+
+  protected validate(): string[] {
+    if (this.targets.length === 0) {
+      return ['No targets added to vpc link'];
+    }
+    return [];
+  }
+
+  private renderTargets() {
+    return this.targets
+      .map(nlb => nlb.vpc?.privateSubnets || [])
```

Comment:
Curious: why only private subnets? (I'm not well versed in NLB, so excuse me if I'm asking the obvious)